riptide-compression/src/test/java/org/zalando/riptide/compression/RequestCompressionPluginTest.java

```diff
@@ -17,94 +19,97 @@
 import org.zalando.riptide.httpclient.ApacheClientHttpRequestFactory;
 import org.zalando.riptide.httpclient.ApacheClientHttpRequestFactory.Mode;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
 import java.util.HashMap;
 import java.util.concurrent.ExecutorService;
-import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
 
-import static com.github.restdriver.clientdriver.ClientDriverRequest.Method.POST;
-import static com.github.restdriver.clientdriver.RestClientDriver.giveResponse;
-import static com.github.restdriver.clientdriver.RestClientDriver.onRequestTo;
+import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.util.Arrays.asList;
 import static java.util.concurrent.Executors.newSingleThreadExecutor;
-import static org.hamcrest.Matchers.emptyString;
-import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.springframework.http.HttpHeaders.CONTENT_ENCODING;
 import static org.zalando.riptide.PassRoute.pass;
+import static org.zalando.riptide.compression.MockWebServerUtil.getBaseUrl;
+import static org.zalando.riptide.compression.MockWebServerUtil.getRecordedRequest;
+import static org.zalando.riptide.compression.MockWebServerUtil.textMockResponse;
 
 class RequestCompressionPluginTest {
 
-    private final ClientDriver driver = GzipClientDriver.create();
+    private final MockWebServer server = new MockWebServer();
+
     private final ExecutorService executor = newSingleThreadExecutor();
 
     @AfterEach
-    void tearDown() throws Exception {
+    void tearDown() {
         executor.shutdown();
-        executor.awaitTermination(10, TimeUnit.SECONDS);
-        driver.verify();
     }
 
     @ParameterizedTest
     @ArgumentsSource(RequestFactorySource.class)
     void shouldCompressRequestBody(final ClientHttpRequestFactory factory) {
-        driver.addExpectation(onRequestTo("/")
-                        .withMethod(POST)
-                        .withHeader("X-Content-Encoding", "gzip") // written by Jetty's GzipHandler
-                        .withBody(equalTo("{}"), "application/json"),
-                giveResponse("", "text/plain"));
+        server.enqueue(textMockResponse(""));
```

Comment:
We cannot use Jetty's `GzipHandler` anymore, so we cannot test header inserted by it. I've tested headers inserted by the plugin and that body was actually compressed (by decompressing it and checking result)